drivers/overlay/overlayutils/utils.go

```diff
@@ -0,0 +1,39 @@
+// Package overlayutils provides utility functions for overlay networks
+package overlayutils
+
+import (
+	"fmt"
+	"sync"
+)
+
+var (
+	vxlanUDPPort uint32
+	mutex        sync.Mutex
+)
+
+func init() {
+	vxlanUDPPort = 4789
+}
+
+// ConfigVxlanUDPPort configures vxlan udp port number.
+func ConfigVxlanUDPPort(vxlanPort uint32) error {
+	mutex.Lock()
+	defer mutex.Unlock()
+	// if the value comes as 0 by any reason we set it to default value 4789
+	if vxlanPort == 0 {
+		vxlanPort = 4789
+	}
+	if vxlanPort < 1024 || vxlanPort > 49151 {
```

Comment:
please add a comment on the reasoning for these values.
The right bound it may be stringent but anyway there is plenty of space left